core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -185,17 +175,23 @@ object ByteBufferMessageSet {
   }
 }
 
+private object OffsetAssigner {
+
+  def apply(offsetCounter: LongRef, size: Int): OffsetAssigner =
+    new OffsetAssigner(offsetCounter.value to offsetCounter.addAndGet(size))
+
+}
+
 private class OffsetAssigner(offsets: Seq[Long]) {
-  val index = new AtomicInteger(0)
+  private var index = 0
 
-  def this(offsetCounter: AtomicLong, size: Int) {
```

Comment:
I replaced the `Atomic` classes because they don't seem to be required. Is this right @becketqin?